shared/actions/kbfs/platform.desktop.js

```diff
@@ -4,63 +4,74 @@ import {shell} from 'electron'
 import * as Constants from '../../constants/config'
 import {Common} from '../../constants/types/keybase-v1'
 import type {AsyncAction} from '../../constants/types/flux'
-import {cleanup} from '../../util/kbfs'
+import path from 'path'
 
-const open = (kbfsPath: string, path: string = '') => { // eslint-disable-line space-infix-ops
-  if (path.startsWith(Constants.defaultPrivatePrefix)) {
-    path = Constants.defaultPrivatePrefix + cleanup(path.slice(Constants.defaultPrivatePrefix.length))
-  } else if (path.startsWith(Constants.defaultPublicPrefix)) {
-    path = Constants.defaultPublicPrefix + cleanup(path.slice(Constants.defaultPublicPrefix.length))
-  } else {
-    path = cleanup(path)
-  }
+function open (openPath: string) {
+  console.log('openItem:', openPath)
+  shell.openItem(openPath)
+}
 
-  shell.openItem(`${kbfsPath}${path}`)
+function openInDefault (openPath: string): AsyncAction {
+  return (dispatch, getState) => new Promise((resolve, reject) => {
+    console.log('openInDefault:', openPath)
+    // Path resolve removes any ..
+    openPath = path.resolve(openPath)
+    // Paths MUST start with defaultKBFSPath
+    if (!openPath.startsWith(Constants.defaultKBFSPath)) {
+      console.warn(`openInKBFS requires ${Constants.defaultKBFSPath} prefix: ${openPath}`)
+      return
+    }
+    open(openPath)
+  })
 }
 
-// Paths MUST start with /keybase/
-export function openInKBFS (path: string = Constants.defaultKBFSPath): AsyncAction {
-  if (!path.startsWith(Constants.defaultKBFSPath)) {
-    console.warn(`ERROR: openInKBFS requires ${Constants.defaultKBFSPath} prefix`)
+function openInWindows (openPath: string = Constants.defaultKBFSPath): AsyncAction {
+  openPath = path.resolve(openPath) // Path resolve removes any ..
+  if (!openPath.startsWith(Constants.defaultKBFSPath)) {
+    console.warn(`openInKBFS requires ${Constants.defaultKBFSPath} prefix: ${openPath}`)
     return () => {}
```

Comment:
`openInDefault` and `openInWindows` differ subtly here: the former bails out asynchronously, the latter bails synchronously. I'd suggest making these consistent to avoid the possibility of subtle bugs in the future.